fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeMetadata.java

```diff
@@ -70,6 +105,81 @@ public List<String> listPartitionNames(String databaseName, String tableName, lo
+                    dbName, tableName, operator);
+        }
+
+        List<RemoteFileDesc> remoteFileDescs = Lists.newArrayList(
+                DeltaLakeRemoteFileDesc.createDeltaLakeRemoteFileDesc(scanTasks));
+        remoteFileInfo.setFiles(remoteFileDescs);
+        return Lists.newArrayList(remoteFileInfo);
+    }
+
+    private void triggerDeltaLakePlanFilesIfNeeded(PredicateSearchKey key, Table table, ScalarOperator operator) {
+        if (!scannedTables.contains(key)) {
+            try (Timer ignored = Tracers.watchScope(Tracers.get(), EXTERNAL, "DELTA_LAKE.processSplit." + key)) {
+                collectDeltaLakePlanFiles(key, table, operator);
+            }
+        }
+    }
+
+    private void collectDeltaLakePlanFiles(PredicateSearchKey key, Table table, ScalarOperator operator) {
+        DeltaLakeTable deltaLakeTable = (DeltaLakeTable) table;
+        Metadata metadata = deltaLakeTable.getDeltaMetadata();
+        Engine engine = deltaLakeTable.getDeltaEngine();
+        SnapshotImpl snapshot = (SnapshotImpl) deltaLakeTable.getDeltaSnapshot();
+
+        StructType schema = metadata.getSchema();
+        Set<String> partitionColumns = metadata.getPartitionColNames();
+
+        List<ScalarOperator> scalarOperators = Utils.extractConjuncts(operator);
+        ScalarOperationToDeltaLakeExpr.DeltaLakeContext deltaLakeContext =
+                new ScalarOperationToDeltaLakeExpr.DeltaLakeContext(schema, partitionColumns);
+        Predicate deltaLakePredicate = new ScalarOperationToDeltaLakeExpr().convert(scalarOperators, deltaLakeContext);
+
+        ScanBuilder scanBuilder = snapshot.getScanBuilder(engine);
+        Scan scan = scanBuilder.withFilter(engine, deltaLakePredicate).build();
+
+        List<Row> files = Lists.newArrayList();
+
+        try (CloseableIterator<FilteredColumnarBatch> scanFilesAsBatches = scan.getScanFiles(engine)) {
+            while (scanFilesAsBatches.hasNext()) {
+                FilteredColumnarBatch scanFileBatch = scanFilesAsBatches.next();
+                try (CloseableIterator<Row> scanFileRows = scanFileBatch.getRows()) {
+                    while (scanFileRows.hasNext()) {
+                        Row scanFileRow = scanFileRows.next();
+                        DeletionVectorDescriptor dv = InternalScanFileUtils.getDeletionVectorDescriptorFromRow(scanFileRow);
+                        if (dv != null) {
+                            ErrorReport.reportValidateException(ErrorCode.ERR_BAD_TABLE_ERROR, ErrorType.UNSUPPORTED,
+                                    "Delta table feature [deletion vectors] is not supported");
+                        }
+                        files.add(scanFileRow);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            LOG.error("Failed to get delta lake scan files", e);
+            throw new StarRocksConnectorException("Failed to get delta lake scan files", e);
+        }
+
+        splitTasks.put(key, files);
```

Comment:
If we query a large table, such as `select count(c) from table`, this seems will consume a lot of memory. I believe Iceberg has the same issue